solo-io/gloo
# Fix validation always accept setting and some flaky tests (#2281)

* Add log lines to debug
* Fix bug
* Fix flaky test
* Reorder for consistency
* Remove logging
* Move to eris errors
* Formatting
* Fix two of three failing tests
* Add changelog
* Remove command from go.mod we didn't need
* Lengthen timeout to remove flakes
* Remove logging line
* Unfocus test
* More detailed comment
* Merge branch 'master' into fix_validation_flake
* Cleaner implementation
* Formatting
* Fixed another test
* This tests works, is just slow/flaky
* Remove outdated setup assertion
* Fail on pending test
* Comment out pending test
* Fix pending test
* Remove extra whitespace
* Replace flaky test with stronger assertion at end of suite
* Test for races in regression tests
* Rename error variables
* Rename special FDS label variables
* Run goimports
* Make flaky test a little less flaky
* Cleanup virtual service after accepted to fix robustness test flake
* Update changelog
* Ensure we're getting the right kind of error
* Handle another flake with invalid resource versions
* Missed part of a copy paste
* Handle resource version flake in robustness test as well
* More comprehensive logging
* Work around one of the two bugs to mostly deflake test :/
* PR feedback, easier to read
* Randomize test specs
* Tests are polluted, can't randomize yet
* Stronger assertions in flaky test
* Use BeEmpty when working with strings
* Increase timeouts
* Merge branch 'master' into fix_validation_flake
* Update changelog
* Temporarily comment out super flaky test and address in follow-up
* PR feedback, use reconciler to handle resource version conflicts
* Randomize suites
* Increase timeout for validation
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2280
resolves https://github.com/solo-io/gloo/issues/2248
